### DISCUSSION TITLE
Cannot call addUpvote/removeUpvote mutations using app-issued user token

### DISCUSSION THREAD

**Discussion by laymonage** [community/community#3968] [Discussions]
Hi,
For the past two months, I've been working on [giscus](https://giscus.vercel.app), a comments widget built on GitHub Discussions. It utilizes the Discussions API to let users sign in with GitHub and leave comments on your website, linking the page with a matching discussion.

Apart from the limitations of the Discussions API (e.g. #3622 and https://github.com/isaacs/github/issues/972), it's been working great. However, when I implemented the upvote toggle feature, I keep getting a `FORBIDDEN: Resource not accessible by integration` error when I call the `addUpvote`/`removeUpvote` GraphQL operations using an app-issued user token (`ghu_xxx...`). Using a personal access token works fine.

The app already has read and write permissions to Discussions. You can try leaving a comment on the website or adding/removing reactions to comments and see that they work.

I've even tried enabling all permissions on the app, but I still got the same error.

Will this be fixed? Thanks!

**Comment by antn** [comment]
Thanks for reporting this! You're right, those mutations don't currently support integrations. I've added this to our engineering todo list.

I can't promise a timeframe for when we'll have a fix for this deployed, but we'll be sure to post an update here once it is!

**Reply by laymonage** [reply]
Hey, sorry for bothering you, but it's been over a year since this was posted. Is this going to be fixed any time soon? 🙏

**Comment by mateoconlechuga** [comment]
Hey any luck on this issue? :)

**Comment by Bestra** [comment]
👋 I've reopened the original issue in our backlog, hopefully we'll get more traction this time around 😅

**Reply by laymonage** [reply]
Thanks. It's been more than six months (two years since the original discussion), Any chance of this happening? 🙂

**Comment by laymonage** [comment]
Bump.

**Comment by Blakomen** [comment]
@antn @Bestra It'd be really useful to get this sorted; is it a complicated engineering adjustment, or is there some way the community can help? It would be helpful to understand how the Discussions API and the features within it are being prioritized.

**Comment by laymonage** [comment]
It's 2025 and almost been four years since I opened this discussion. It's very disappointing to see that GitHub has practically stopped developing Discussions and left most-upvoted requests (like #2971) unanswered, even a small bug fix like this one. Can we at least get an update on why this hasn't been fixed? Is there a particular technical challenge, or is it just nobody at GitHub cares?

**Reply by IIiiLlliI** [reply]
HaHa,Hello laymonage, I 'm a Chinese Student Develop, I 'm trying to imitate your project giscus.Because your project doesn't seem to support custom styles, embedding it in my blog doesn't look good.Then I also encountered the same problem. Fortunately, you had already tried out the method of checking all permissions for me.😊

**Comment by munrocket** [comment]
Any progress?

**Comment by anasyounis513** [comment]
Thanks for the detailed explanation — I’ve run into a similar issue with the Discussions API when using app-issued tokens. From what I understand, the addUpvote and removeUpvote mutations currently aren’t fully supported for GitHub Apps, even if all permissions are enabled.
It seems like a limitation on GitHub’s side rather than your implementation. Hopefully, they extend full reaction support to apps soon. Until then, using a PAT is the only reliable workaround, but not ideal for production.
Appreciate your work on Giscus — it's an awesome tool!

**Comment by actforjason** [comment]
The lack of this feature overshadows the entire.
